### ISSUE TITLE
Update-metadata failure

### ISSUE THREAD

**Issue by marcmengel** [fermitools/metacat#1]
[from @rlcee in [upstream/#41](https://github.com/ivmfnal/metacat/issues/41) ]

In metacat-3.42.0

```
metacat file update-meta -f rlc:etc.rlc.dh_test.000.000.txt '{ "mm.testb" : true, "mm.tests" : ["seqa"] }'

Application error
update_many() got an unexpected keyword argument 'do_commit'
Traceback (most recent call last):
  File "/home/dbidata3/multiserver/webpie/webpie/WPApp.py", line 644, in wsgi_call
    response = root_handler._handle_request(request, "", path_down, args)
  File "/home/dbidata3/multiserver/webpie/webpie/WPApp.py", line 437, in _handle_request
    return subhandler._handle_request(request, path + "/" + word, path_down, args)
  File "/home/dbidata3/multiserver/webpie/webpie/WPApp.py", line 435, in _handle_request
    return subhandler(request, relpath, **args)
  File "./metacat_server_generic/server/common_handler.py", line 93, in decorated
    try:    out = method(*params, **agrs)
  File "./metacat_server_generic/server/data_handler.py", line 1185, in update_file_meta
    return self.__update_meta_bulk(db, user, data["metadata"], data["mode"], ids=by_fid, names=by_namespace_name)
  File "./metacat_server_generic/server/data_handler.py", line 1005, in __update_meta_bulk
    DBFile.update_many(db, file_set, do_commit=True)
  File "./metacat_server_generic/lib/metacat/common/dbbase.py", line 18, in decorated
    return method(first, *params, transaction=transaction, **args)
TypeError: update_many() got an unexpected keyword argument 'do_commit'
```

**Comment by marcmengel** [comment]
It looks like the failure is either in [the call](https://github.com/marcmengel/metacat/blob/main/webserver/data_handler.py#L1005) where we call update_many, or in [the definition](https://github.com/marcmengel/metacat/blob/main/metacat/db/dbobjects2.py#L642) where the arguments don't accept a do_commit flag... since update_many is decorated with `@transactioned` I must assume it *always* commits, and since no-one else calls update_many, (according to git grep) I think we can just drop the flag in the call.

**Comment by marcmengel** [comment]
Okay, so getting rid of that flag gets us furhter; now we get:
```
<html><body><h2>Application error</h2>
            <h3>list index out of range</h3>
            <pre>
Traceback (most recent call last):
  File "/home/dbidata3/multiserver/webpie/webpie/WPApp.py", line 644, in wsgi_call
    response = root_handler._handle_request(request, "", path_down, args)
  File "/home/dbidata3/multiserver/webpie/webpie/WPApp.py", line 437, in _handle_request
    return subhandler._handle_request(request, path + "/" + word, path_down, args)
  File "/home/dbidata3/multiserver/webpie/webpie/WPApp.py", line 435, in _handle_request
    return subhandler(request, relpath, **args)
  File "./metacat_server_test/server/common_handler.py", line 93, in decorated
    try:    out = method(*params, **agrs)
  File "./metacat_server_test/server/data_handler.py", line 1185, in update_file_meta
    return self.__update_meta_bulk(db, user, data["metadata"], data["mode"], ids=by_fid, names=by_namespace_name)
  File "./metacat_server_test/server/data_handler.py", line 1005, in __update_meta_bulk
    DBFile.update_many(db, file_set)
  File "./metacat_server_test/lib/metacat/common/dbbase.py", line 18, in decorated
    return method(first, *params, transaction=transaction, **args)
  File "./metacat_server_test/lib/metacat/db/dbobjects2.py", line 649, in update_many
    transaction.executemany("""
  File "/home/dbidata3/multiserver/wsdbtools/wsdbtools/transaction.py", line 37, in executemany
    self.Cursor.execute(*params, **args)
IndexError: list index out of range
</pre>
            </body>
            </html>
```

**Comment by marcmengel** [comment]
So that in turn led to a bug in wsdbtools, which has an easy [fix](https://github.com/ivmfnal/wsdbtools/pull/1)

**Comment by marcmengel** [comment]
So @rlcee  this is working in the "hypot" instance on dbweb5 now; I need to rsync it around to the other hosts, and then schedule an update to the production instances.

**Comment by marcmengel** [comment]
So I believe this is fixed in version 3.42.3, which is what mu2e is now running.

**Comment by rlcee** [comment]
I confirm that update metadata is now working for me.  Thanks very much, you can close this issue.
